ndsl/comm/communicator.py

```diff
@@ -93,6 +94,27 @@ def _device_synchronize():
         # this is a method so we can profile it separately from other device syncs
         device_synchronize()
 
+    def _create_all_reduce_quantity(
+        self, input_metadata: QuantityMetadata, input_data
+    ) -> Quantity:
+        """Create a Quantity for all_reduce data and metadata"""
+        all_reduce_quantity = Quantity(
+            input_data,
+            dims=input_metadata.dims,
+            units=input_metadata.units,
+            origin=tuple([0 for dim in input_metadata.dims]),
```

Comment:
We are going to have replicate `origin` & `extent` or else we are going to alter the offset of the computation going downward

Reply:
Done. Set `origin=input_metadata.origin` and added `extent=input_metadata.extent`.